### PR TITLE
include: dt-bindings: pinctrl_stm32: Fix for pinctrl GPIO-out pin confguration

### PR DESCRIPTION
**Pull request by LuxLuthor**
Added information about pin mode into Z_PINCTRL_STM32_PINCFG_INIT. _pinctrl_configure_pins()_ does not need to figure out GPIO pin direction now.

### CODE REVIEW

File: soc/arm/st_stm32/common/pinctrl_soc.h [hunk 1]
```diff
@@ -46,13 +46,14 @@ typedef struct pinctrl_soc_pin {
...
  * @brief Definitions used to initialize fields in #pinctrl_pin_t
  */
-#define STM32_NO_PULL     0x0
-#define STM32_PULL_UP     0x1
-#define STM32_PULL_DOWN   0x2
-#define STM32_PUSH_PULL   0x0
+#define STM32_NO_PULL	  0x0
+#define STM32_PULL_UP	  0x1
+#define STM32_PULL_DOWN	  0x2
+#define STM32_PUSH_PULL	  0x0
```
Comment (erwango) [inline soc/arm/st_stm32/common/pinctrl_soc.h:52]: Make this a dedicated commit.

**Review by erwango — CHANGES_REQUESTED** [review]
Thanks, better solution than https://github.com/zephyrproject-rtos/zephyr/pull/53211.
Though, please provide more comments, either in the commit message or `pinctrl_configure_pins`.

Then, please fix your commit title.

File: soc/arm/st_stm32/common/pinctrl_soc.h [hunk 2]
```diff
@@ -46,13 +46,14 @@ typedef struct pinctrl_soc_pin {
...
 #define STM32_OPEN_DRAIN  0x1
 #define STM32_OUTPUT_LOW  0x0
 #define STM32_OUTPUT_HIGH 0x1
+#define STM32_MODE_OUTPUT 0x1
```
Comment (erwango) [inline soc/arm/st_stm32/common/pinctrl_soc.h:56]: This conflicts with F1 dt-binding definition

**Review by erwango — APPROVED** [review]

**Comment by erwango** [comment]
@LuxLuthor Approved, but please don't remove reviewers. You need 2 positive reviews for the PR to be merged.

**Comment by LuxLuthor** [comment]
> @LuxLuthor Approved, but please don't remove reviewers. You need 2 positive reviews for the PR to be merged.

Thanks. I have only clicked "re-request review" and it somehow have removed all other reviewers. How can I bring them back :) ?

**Review by carlescufi — APPROVED** [review]
